src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -668,6 +674,20 @@ protected DynamoModel(IStartConfiguration config)
             {
                 // Do nothing for now
             }
+
+            // these configuration options are incompatible, one requires loading ADP binaries
+            // the other depends on not loading those same binaries.
+
+            if (areAnalyticsDisabledFromConfig && disableADP)
+            {
+                throw new ConfigurationException("Incompatible configuration: could not start Dynamo with both [Analytics disabled] and [ADP disabled] config options enabled");
+            }
+
+            if(IsTestMode && disableADP)
+            {
+                this.Logger.Log("Incompatible configuration: [IsTestMode] and [ADP disabled] ");
```

Comment:
what does this case mean?

Reply:
this is the case where testmode is active (during most tests) and some test might try to disable ADP - but we can't guarantee that we have access to the core sdks in different testing scenarios, and I could not find a nice way to mock this dependency, so thought it would be valuable to log the case.

Reply:
That is exactly what I was going to mention. We always bail out during test mode so I dont think ADP will be loaded at all? https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Models/DynamoModel.cs#L1429

Reply:
we don't *need* it - but I think it might help avoid wasted time. It would be nice if Dynamo provided feedback on incompatible options since to a client it's not obvious what headless and testmode actually control.

Reply:
Thanks @mjkkirschner If that's the case, it seems this warning can be moved into beginning of `AnalyticsService.Start()`

Reply:
yes, can do.